### ISSUE TITLE
Failed to build iOS app

### ISSUE THREAD

**Issue by edenko** [DenisovAV/flutter_gemma#5]
![image](https://github.com/DenisovAV/flutter_gemma/assets/58899690/15fdcf2d-7f5f-46d5-8500-29f58852b4ef)

Is there another way? 
I added the settings to the main, but what am I missing?

------------------------------------------------------------------------------------

import Foundation
import MediaPipeTasksGenAI

final class InferenceModel {
    private var inference: LlmInference!

    init(maxTokens: Int, temperature: Float, randomSeed: Int, topK: Int) {
        let fileManager = FileManager.default
        let documentDirectory = fileManager.urls(for: .documentDirectory, in: .userDomainMask).first!
        let filePath = documentDirectory.appendingPathComponent("model.bin").path

        do {
            let llmOptions = LlmInference.Options(modelPath: filePath)
            llmOptions.maxTokens = maxTokens
            llmOptions.temperature = temperature
            llmOptions.randomSeed = randomSeed
            llmOptions.topk = topK
            self.inference = try LlmInference(options: llmOptions)
        } catch {
            print("Failed to initialize LlmInference: \(error)")
        }
    }

    func generateResponse(prompt: String) throws -> String {
        return try inference.generateResponse(inputText: prompt)
    }
    
    func generateResponseAsync(prompt: String, progress: @escaping (_ partialResponse: String?, _ error: Error?) -> Void, completion: @escaping (() -> Void)) throws {
        do {
            try inference.generateResponseAsync(inputText: prompt, progress: progress, completion: completion)
        } catch {
            throw error
        }
    }
}

This is a build

**Comment by DenisovAV** [comment]
Do you run an example or your own project? Could you tell me your XCode and MacOS versions? I have no such error.

**Comment by edenko** [comment]
I am running my own project. 
My Xcode version is 15.4 (15F31d) and macOS version is 14.1.2.

**Comment by hypen-flutter** [comment]
Same issue. Mac 15.0, Xcode 16.0

**Comment by DenisovAV** [comment]
please check 0.2.2 version

**Comment by dsx1986** [comment]
I am using 0.2.2 and I got similar error: 


Swift Compiler Error (Xcode): Value of type 'LlmInference.Options' has no member 'temperature'
/Users/simon/.pub-cache/hosted/pub.dev/flutter_gemma-0.2.2/ios/Classes/InferenceModel.swift:13:19


Swift Compiler Error (Xcode): Value of type 'LlmInference.Options' has no member 'randomSeed'
/Users/simon/.pub-cache/hosted/pub.dev/flutter_gemma-0.2.2/ios/Classes/InferenceModel.swift:14:19


Swift Compiler Error (Xcode): Value of type 'LlmInference.Options' has no member 'topk'
/Users/simon/.pub-cache/hosted/pub.dev/flutter_gemma-0.2.2/ios/Classes/InferenceModel.swift:15:19

**Comment by DenisovAV** [comment]
I looks like the rootcause in an incompatibility of mediapipe version, I fixed the version at 0.2.3, please try

**Comment by DenisovAV** [comment]
Can you share Podfile.lock with me, to check version number of MediaPipeTasksGenAI

**Comment by dsx1986** [comment]
flutter_gemma:
    dependency: "direct main"
    description:
      name: flutter_gemma
      sha256: f0ebe363161599fdc69df47cfc07bec317e20a502a47959aecd14e1c9cfc17ca
      url: "https://pub.dev"
    source: hosted
    version: "0.2.3"
    
    
    Installing MediaPipeTasksGenAI (0.10.16)
Installing MediaPipeTasksGenAIC (0.10.16)

**Comment by DenisovAV** [comment]
I'm going to update the version of MediaPipeTasksGenAI to the latest one, but it will be in a couple of weeks, after downloading model implemetation

**Comment by dsx1986** [comment]
> It should be MediaPipeTasksGenAI (0.10.12) as in configuration, try to do flutter clean, delete Podfile.lock, run flutter pub get again and check. Probably this is a cached version

Yes, I deleted local pub cache and Cocoa pods cache, and build foldler, Podfile.lock. It is still installing 0.10.16 version:

Installing MediaPipeTasksGenAI (0.10.16)
Installing MediaPipeTasksGenAIC (0.10.16)

**Comment by DenisovAV** [comment]
I hard fixed the version in 0.2.4 could you please try?

**Comment by dsx1986** [comment]
I got error when build android:


ERROR: /Users/simon/git/diogenesaichatbot/build/flutter_secure_storage/intermediates/merged_res/release/mergeReleaseResources/values/values.xml:194: AAPT: error: resource android:attr/lStar not found.


your flutter_secure_storage 4.2.1  is too low.(9.2.2 available)

can you fix it?

**Comment by dsx1986** [comment]
yes, it is working. it has some version conflict with flutter_secure_storage but I avoid using flutter_secure_storage and now it is working on android.

**Comment by dsx1986** [comment]
ios failed when try to init the model. it is most probably due to model path issue

**Comment by dsx1986** [comment]
ios run successful. I put my model in cloud and download it to device when need it. Right now we have to rename model name to model.bin to make it work, can you change the interface to allow file name

**Comment by dsx1986** [comment]
cpu version is too slow on physical device. Is there option to enable GPU?

**Comment by DenisovAV** [comment]
In the next version, you will be able to use a model from the cloud out of the box, I will publish it today. Regarding GPU,  try to use GPU model. 
<img width="1476" alt="image" src="https://github.com/user-attachments/assets/5004178c-ea3b-4995-812b-ec08c567b603">

**Comment by dsx1986** [comment]
thanks. I have gpu model and tried. the gpu model crashed while cpu version works

**Comment by DenisovAV** [comment]
I checked my physical device, everything is fine, I have iPhone 14 Pro. and btw check 0.3.0, there is implemented upload model feature

**Comment by DenisovAV** [comment]
@dsx1986 can you check with the last version of the plugin?
